fiftyone/core/dataset.py

```diff
@@ -4583,12 +4583,15 @@ def _get_random_characters(n):
 def _list_datasets(include_private=False):
     conn = foo.get_db_conn()
 
+    # We don't want an error here if `name == None`
+    _sort = lambda l: sorted(l, key=lambda x: (x is None, x))
+
     if include_private:
-        return sorted(conn.datasets.distinct("name"))
+        return _sort(conn.datasets.distinct("name"))
 
     # Datasets whose sample collections don't start with `samples.` are private
     # e.g., patches or frames datasets
-    return sorted(
+    return _sort(
         conn.datasets.find(
             {"sample_collection_name": {"$regex": "^samples\\."}}
         ).distinct("name")
```

Comment:
Nitpick. I prefer to avoid repeated code in situations like this.
```py
    query = {} # {"name": {"$ne": None}} if you want to let Mongo do your filtering
    if not include_private:
        # Datasets whose sample collections don't start with `samples.` are private
        # e.g., patches or frames datasets
        query["sample_collection_name"] = {"$regex": "^samples\\."}}    

    # We don't want an error here if `name == None`    
    return sorted(
        conn.datasets.find(query).distinct("name"),
        key=lambda x: (x is None, x) # remove this line if you let Mongo filter
    )
```